### PR TITLE
Use Mozilla bug REST API. Fixes #7

### LINKED ISSUE #7: Use Bugzilla API instead of CORs fetching document
https://bmo.readthedocs.io/en/latest/api/core/v1/bug.html#get-bug

### PR DESCRIPTION
**Pull request by ebidel**
R: @PaulKinlan

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -131,18 +131,38 @@ class WebKitBug extends Bug {
...
 
-  findStatus(doc) {
-    const statusEl = doc.querySelector('#static_bug_status');
-    if (statusEl) {
-      return statusEl.textContent.replace('\n', '')
-                                 .replace(/ +(?=)/g, ' ').trim();
+  fetchPage(url=this.url) {
+    let match = url.match(/\?id=(.*)$/);
+    if (!match) {
+      return Promise.reject('Could not find Mozilla bug id.');
```
Comment (PaulKinlan) [inline js/app.js]: Minor: This reads like the bug can't be found in the DB and not in the URL.  'Could not find Mozilla bug id in query string'

File: js/app.js [hunk 2]
```diff
@@ -131,18 +131,38 @@ class WebKitBug extends Bug {
...
+    this.needsCors = false;
   }
 
-  findStatus(doc) {
-    const statusEl = doc.querySelector('#static_bug_status');
-    if (statusEl) {
-      return statusEl.textContent.replace('\n', '')
-                                 .replace(/ +(?=)/g, ' ').trim();
+  fetchPage(url=this.url) {
+    let match = url.match(/\?id=(.*)$/);
```
Comment (PaulKinlan) [inline js/app.js]: This will match id=, change to .+ so that we know there is a bug id.

File: js/app.js [hunk 3]
```diff
@@ -258,7 +278,7 @@ function populateBugStatus(items) {
         break;
       case 'Mozilla':
         let mozillaBug = new MozillaBug(item.link);
-        var p = mozillaBug.fetchPage().then(doc => mozillaBug.findStatus(doc)).then(status => {
+        var p = mozillaBug.fetchPage().then(json => mozillaBug.findStatus(json)).then(status => {
           status = status.toUpperCase();//.replace(/RESOLVED(.*)?/, 'FIXED');
```
Comment (PaulKinlan) [inline js/app.js]: Remove comment?

File: js/app.js (hunk 2 above)
Comment (ebidel) [inline js/app.js]: Done

File: js/app.js (hunk 1 above)
Comment (ebidel) [inline js/app.js]: "Could not find bug id Mozilla bug link.'"

File: js/app.js (hunk 3 above)
Comment (ebidel) [inline js/app.js]: Done

**Comment by ebidel** [comment]
PTAL

**Pull request merged**